### PR TITLE
Add saving coils

### PR DESCRIPTION
**Pull request by gbaranski**
@stasbar can You check that out? I'm getting error 'permission denied'.

### CODE REVIEW

**Comment by stasbar** [comment]
@gbaranski Please review my changes, you still need to create some Models where the user can enter Coil | Liquid names, descriptions

**Comment by gbaranski** [comment]
@stasbar done

**Comment by gbaranski** [comment]
@stasbar done

File: src/pages/coil/CoilCalculator.tsx [hunk 1]
```diff
@@ -70,6 +71,8 @@ const CoilCalculator: React.FC<CoilCalculatorProps> = props => {
   const [helpModalVisibile, setHelpModalVisible] = useState(false);
   const [slider, setSlider] = useState<Carousel>();
   const [saveModalVisible, setSaveModalVisible] = useState(false);
+  const [coilName, setCoilName] = useState('');
+  const [coilDescription, setCoilDescription] = useState('');
```
Comment (stasbar) [inline src/pages/coil/CoilCalculator.tsx:75]: move them to SaveModal.

File: src/pages/coil/CoilCalculator.tsx [hunk 2]
```diff
@@ -158,12 +161,18 @@ const CoilCalculator: React.FC<CoilCalculatorProps> = props => {
...
-      <SaveCoilModal
+      <SaveModal
         visible={saveModalVisible}
         setVisible={setSaveModalVisible}
-        coil={coil}
-        userUid={user ? user.uid : ''}
-        userName={user ? user.name : ''}
+        setDescription={setCoilDescription}
+        setName={setCoilName}
+        save={async () => {
```
Comment (stasbar) [inline src/pages/coil/CoilCalculator.tsx:169]: `save={ async (name, description) => {`

File: src/pages/coil/CoilCalculator.tsx [hunk 3]
```diff
@@ -158,12 +161,18 @@ const CoilCalculator: React.FC<CoilCalculatorProps> = props => {
...
     <Card style={{ height: '100%' }}>
-      <SaveCoilModal
+      <SaveModal
         visible={saveModalVisible}
         setVisible={setSaveModalVisible}
-        coil={coil}
-        userUid={user ? user.uid : ''}
-        userName={user ? user.name : ''}
+        setDescription={setCoilDescription}
+        setName={setCoilName}
```
Comment (stasbar) [inline src/pages/coil/CoilCalculator.tsx:168]: remove

File: src/pages/coil/CoilCalculator.tsx [hunk 4]
```diff
@@ -158,12 +161,18 @@ const CoilCalculator: React.FC<CoilCalculatorProps> = props => {
...
+        save={async () => {
+          if (user && user.uid && user.name) {
+            saveCoil(coil, new Author(user.uid, user.name), coilName, coilDescription);
```
Comment (stasbar) [inline src/pages/coil/CoilCalculator.tsx:171]: saveCoil(coil, new Author(user.uid, user.name), name, description);

File: src/pages/liquid/LiquidBlender.tsx [hunk 5]
```diff
@@ -92,12 +96,23 @@ const LiquidBlender: React.FC<LiquidBlenderProps> = ({
...
-      <SaveLiquidModal
+      <SaveModal
         visible={saveModalVisible}
         setVisible={setSaveModalVisible}
-        currentLiquid={currentLiquid}
-        userUid={user ? user.uid : ''}
-        userName={user ? user.name : ''}
+        setDescription={setLiquidDescription}
+        setName={setLiquidName}
+        save={async () => {
```
Comment (stasbar) [inline src/pages/liquid/LiquidBlender.tsx:104]: `save={ async (name, description) => {`

File: src/pages/liquid/LiquidBlender.tsx [hunk 6]
```diff
@@ -92,12 +96,23 @@ const LiquidBlender: React.FC<LiquidBlenderProps> = ({
...
+          if (user && user.uid && user.name) {
+            saveLiquid(
+              currentLiquid,
+              new Author(user.uid, user.name),
+              liquidName,
+              liquidDescription,
```
Comment (stasbar) [inline src/pages/liquid/LiquidBlender.tsx:110]: `saveLiquid(liquid, new Author(user.uid, user.name), name, description);`

File: src/pages/liquid/LiquidBlender.tsx [hunk 7]
```diff
@@ -66,6 +68,8 @@ const LiquidBlender: React.FC<LiquidBlenderProps> = ({
   user,
 }) => {
   const [saveModalVisible, setSaveModalVisible] = useState(false);
+  const [liquidName, setLiquidName] = useState('');
+  const [liquidDescription, setLiquidDescription] = useState('');
```
Comment (stasbar) [inline src/pages/liquid/LiquidBlender.tsx:72]: remove

File: src/pages/coil/CoilCalculator.tsx [hunk 8]
```diff
@@ -158,12 +161,18 @@ const CoilCalculator: React.FC<CoilCalculatorProps> = props => {
...
+          } else {
+            throw new Error();
```
Comment (stasbar) [inline src/pages/coil/CoilCalculator.tsx:173]: set reason in Error constructor
`new Error('Can not save with undefined user ')`

File: src/pages/liquid/LiquidBlender.tsx [hunk 9]
```diff
@@ -92,12 +96,23 @@ const LiquidBlender: React.FC<LiquidBlenderProps> = ({
...
+        save={async () => {
+          if (user && user.uid && user.name) {
+            saveLiquid(
+              currentLiquid,
+              new Author(user.uid, user.name),
+              liquidName,
+              liquidDescription,
+            );
+          } else {
+            throw new Error();
```
Comment (stasbar) [inline src/pages/liquid/LiquidBlender.tsx:113]: set reason in Error constructor
`new Error('Can not save with undefined user ')`

File: src/pages/liquid/LiquidBlender.tsx [hunk 10]
```diff
@@ -92,12 +96,23 @@ const LiquidBlender: React.FC<LiquidBlenderProps> = ({
...
     <div>
-      <SaveLiquidModal
+      <SaveModal
         visible={saveModalVisible}
         setVisible={setSaveModalVisible}
-        currentLiquid={currentLiquid}
-        userUid={user ? user.uid : ''}
-        userName={user ? user.name : ''}
+        setDescription={setLiquidDescription}
+        setName={setLiquidName}
```
Comment (stasbar) [inline src/pages/liquid/LiquidBlender.tsx:103]: remove

File: src/components/saveToDatabaseModal/index.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+import { Modal, Input } from 'antd';
+import FormItem from 'antd/lib/form/FormItem';
+import { FormattedMessage } from 'umi-plugin-react/locale';
+
+export default function SaveModal(props: any) {
+  const [isLoading, setLoading] = useState(false);
+
+  const handleOk = async () => {
+    setLoading(true);
+    await props.save();
```
Comment (stasbar) [inline src/components/saveToDatabaseModal/index.tsx:11]: await props.save(name, description);

File: src/components/saveToDatabaseModal/index.tsx [hunk 12]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import { Modal, Input } from 'antd';
+import FormItem from 'antd/lib/form/FormItem';
+import { FormattedMessage } from 'umi-plugin-react/locale';
+
+export default function SaveModal(props: any) {
+  const [isLoading, setLoading] = useState(false);
+
```
Comment (stasbar) [inline src/components/saveToDatabaseModal/index.tsx:8]: const [name, setName] = useState('');
const [description, setDescription] = useState('');

File: src/pages/coil/CoilCalculator.tsx (hunk 1 above)
Comment (gbaranski) [reply src/pages/coil/CoilCalculator.tsx:75]: but how then i can refer to a child property?

File: src/pages/coil/CoilCalculator.tsx (hunk 2 above)
Comment (gbaranski) [reply src/pages/coil/CoilCalculator.tsx:169]: but thats taking data from function scope state

**Pull request merged**